src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -46,9 +42,13 @@ protected function __construct(string $namespace = '', int $defaultLifetime = 0)
             function ($key, $value, $isHit) use ($defaultLifetime) {
                 $item = new CacheItem();
                 $item->key = $key;
+                $item->defaultLifetime = $defaultLifetime;
+                // If structure is in TagAware format, return as is (no value and not a hit)
+                if (\is_array($value) && \array_key_exists('value', $value)) {
```

Comment:
I'm really unsure about this check: this doesn't look specific enough to me. Not we need to do anything IMHO.

Reply:
Not that we have namespace separation we can drop it